core/Pos/Binary/Core/Address.hs

```diff
@@ -138,9 +138,14 @@ instance Bi (Attributes AddrAttributes) where
             , aaStakeDistribution = BootstrapEraDistr
             }
         go n v acc =
+            -- [CSL-1493] Should we:
+            -- 1. fail explicitly here (by using risky unsafe functions)
+            -- 2. Yield `Nothing` in case we cannot deserialize `v`
+            -- 3. Make `go` and thus `decodeAttributes` take an impure `updater`
+            --    function (which runs in the `Decoder` monad).
```

Comment:
Great! Will go with (3) then and report back if there are any unforeseen hiccups.